### ISSUE TITLE
Consistency in FaceQuadratureInterpolator and pa_data

### ISSUE THREAD

**Issue by artv3** [mfem/mfem#2865]
When applying FaceQuadratureInterpolator to a field it returns the data in the order `<quad, dim, nf>`; while pa_data will organize it in terms of `<dim, quad, nf>` (ex. DG trace pa setup kernels)

QuadratureInterpolator has a method to switch between layouts, 
``SetOutputLayout(QVectorLayout::byVDIM); ``

Can something like that be introduced to FaceQuadratureInterpolator to maintain consistency?

**Comment by YohannDudouit** [comment]
@artv3 Could you be a bit more specific on which kernels you are thinking of?
Here is one in `quadinterpolator_face.cpp:FaceQuadratureInterpolator::Eval3D`:
```c++
nor(q1,q2,0,f) = n[0]/norm;
```
Here is another in `bilininteg_dgtrace_pa.cpp:PADGTraceSetup3D`:
```c++
qd(q1,q2,0,0,f) = w*( alpha/2 * dot + beta * abs );
```
They both seem to use `<quad, dim, nf>` ordering.

**Comment by artv3** [comment]
yes, right above that when you take the velocity

https://github.com/mfem/mfem/blob/5308d285f732c45f3c381bcee415581fd34b1cb2/fem/bilininteg_dgtrace_pa.cpp#L95-L97

the ordering is `<dim, quad, nf>` so if the user wants to provide the velocity we have to flip the layout from what FaceQuadrature interpolator gives which is `<quad, dim, nf>` to `<dim, quad, nf>`

**Comment by artv3** [comment]
I have an application where I compute the velocity and I would like to pass that to mfem in a container.

**Comment by YohannDudouit** [comment]
Ok, I can change that, the only question is if it will impact codes using `VectorQuadratureFunctionCoefficient`, do you use that? Technically it may break backward compatibility.

**Comment by artv3** [comment]
I think we just need to add a method to switch between the layouts in `FaceQuadratureInterpolator`, that shouldn't break things.

**Comment by camierjs** [comment]
We could still do something like in [qinterp eval](https://github.com/mfem/mfem/blob/master/fem/qinterp/eval.hpp) for the layout if we can't wait for a more generic way to handle it.

**Comment by v-dobrev** [comment]
This can be done as part of #1656. The idea will be to make sure we can evaluate `GridFunctionCoefficient`s on face quadrature spaces, so that the resulting data is ordered as needed by the face quadrature space which will then be used inside face-based PA kernels.

**Comment by v-dobrev** [comment]
This issue will be addressed as part of the unification of #1656 and #2891, so I'll close it.
